src/puffin.rs

```diff
@@ -0,0 +1,45 @@
+use puffin::{GlobalProfiler, NanoSecond, ScopeDetails, StreamInfo, ThreadInfo};
+
+use crate::GpuTimerQueryResult;
+
+/// Visualize the query results in a `puffin::GlobalProfiler`.
+pub fn output_frame_to_puffin(profiler: &mut GlobalProfiler, query_result: &[GpuTimerQueryResult]) {
+    let mut stream_info = StreamInfo::default();
+    recusive_profile_gpu_frame(profiler, &mut stream_info, query_result, 0);
```

Comment:
```suggestion
    output_frame_to_puffin_recursive(profiler, &mut stream_info, query_result, 0);
```

Reply:
Good point, this function's name didn't really explain what it did. However I think `collect_stream_info_recursive` might be a more useful name. This function doesn't actually output anything to puffin, it just does the intermediate step of adding things to a `StreamInfo`.